### DISCUSSION TITLE
JWT access_token

### DISCUSSION THREAD

**Discussion by BenDurkin** [Jericho/ZoomNet#145] [Q&A]
To access cloud recordings there is a property called download_url, but this is password protected. To access it directly you need to append a parameter called access_token which contains the JWT. 

Is there anyway to get the JWT as a string using the JwtConnectionInfo object?

Thanks

**Comment by Jericho** [comment]
The functionality to download a recording file is already built-in ZoomNet, so no need to hand craft the URL to download the file. However, I just realized that the `DownloadFileAsync` is on the concrete `CloudRecordings` class only and is missing from the `ICloudRecordings` interface. I intended for this method to be available like so:

```csharp
var fileStream = await client.CloudRecordings.DownloadFileAsync(recordingFile, cancellationToken).ConfigureAwait(false);
```

but it's not available due to the fact that it's not on the interface. I will resolve this issue and publish an update ASAP.

Short term work around is to cast the `CloudRecordings` to the concrete class like so:

```csharp
var fileStream = await ((CloudRecordings)client.CloudRecordings).DownloadFileAsync(recordingFile, cancellationToken).ConfigureAwait(false);
```

**Comment by Jericho** [comment]
Version 0.33.0 published with a fix for the missing method on the ICloudRecordings interface.

Let me know if this answers your question.
